libs/event-manager/web-client/shell/src/lib/shell.component.ts

```diff
@@ -83,11 +89,40 @@ import {
   providers: [ConfigStore],
 })
 export class ShellComponent implements OnInit {
+  readonly loggedIn = this._loginStore.connected$;
+
   constructor(
     private readonly _hdConnectionStore: ConnectionStore,
     private readonly _hdWalletStore: WalletStore,
+    private readonly _loginStore: LoginStore,
+    private readonly _matSnackBar: MatSnackBar,
+    private readonly _router: Router,
     @Inject(ENVIRONMENT_CONFIG) private environment: EnvironmentConfig
-  ) {}
+  ) {
+    _hdWalletStore.publicKey$.subscribe(async (publicKey) => {
```

Comment:
This is a potential memory leak, to prevent it you have to make sure to unsubscribe when the component is destroyed. [how to unsubscribe](https://blog.bitsrc.io/6-ways-to-unsubscribe-from-observables-in-angular-ab912819a78f)